### PR TITLE
HPCC-27281 Deploying embedded jar file via githook

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: ecl/hqlcpp/hqlres.cpp [hunk 1]
```diff
@@ -171,7 +171,8 @@ void ResourceManager::addManifestFile(const char *filename, ICodegenContextCallb
 {
     StringBuffer fileContents;
     StringBuffer strippedFileContents;
-    fileContents.loadFile(filename, false);
+    Owned<IFile> manifestFile = createIFile(filename);
+    fileContents.loadFile(manifestFile);
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlres.cpp:175]: There is a similar call in
ResourceManifest::ResourceManifest

File: ecl/eclcc/eclcc.cpp [hunk 2]
```diff
@@ -853,7 +853,8 @@ void gatherResourceManifestFilenames(EclCompileInstance & instance, StringArray
             getFullFileName(filename, true).append(".manifest");
             if (filenames.contains(filename))
                 continue;
-            if (checkFileExists(filename))
+            Owned<IFile> manifest = createIFile(filename);
+            if (manifest->exists())
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:857]: How about the call in  ResourceManifest::checkResourceFilesExist?
Also potentially isManifestFileValid, although much less likely.

File: common/remote/hooks/git/gitfile.cpp [hunk 3]
```diff
@@ -263,6 +263,15 @@ class GitRepositoryFile : implements IFile, public CInterface
     {
         if (!isExisting)
             return (offset_t) -1;
+        if (fileSize != (offset_t) -1)
+            return fileSize;
+        git_blob *blob = nullptr;
```
Comment (ghalliday) [inline common/remote/hooks/git/gitfile.cpp:268]:
```
if (isDir)
   return 0;

```
either here or set fileSize in the GitRepositoryFile constructor.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman one comment

File: common/remote/hooks/git/gitfile.cpp (hunk 3 above)
Comment (richardkchapman) [reply common/remote/hooks/git/gitfile.cpp]: Fixed and repushed

**Review by ghalliday — APPROVED** [review]

**Pull request merged**